### ISSUE TITLE
DLC in COLAB using Training Data coming from both a Mac and Windows OS (how to combine data properly?)

### ISSUE THREAD

**Issue by pkudzia** [DeepLabCut/DeepLabCut#1411]
**Describe the problem**
I have two team members labelling data . One is on a windows computer and one on a mac. I combined the two folders and changed all the appropriate names so that the .config file has access to all the data. 

Then i changed this command in COLAB to include the windows to linux conversion :
deeplabcut.create_training_dataset(path_config_file, net_type='resnet_50', Shuffles=[SHUF], augmenter_type='imgaug',windows2linux=True)

When that command was not available it would shoot the common error that people were getting when loading windows files.

Now everything works fine i.e I can make a training set but as you can see from the screenshot below the training data is separated for the windows files and the mac files. The .mat file only contains the training data coming from the mac (and i assume also the pickle file). So when I run the training it is not including all the available data. 

Any tips on how to merge these successfully for training in COLAB would be awesome. 

**Screenshots**
![Screen Shot 2021-07-14 at 11 54 52 AM](https://user-images.githubusercontent.com/15024880/125677056-1474a05e-627e-4572-a74d-a85eec793325.png)

**Comment by jeylau** [comment]
Hey! The windows.csv and h5 files are merely backups of your data prior to conversion, so all your data should indeed be found in CollectedData_Serena.h5 (you could inspect the csv to verify) 😊

**Comment by pkudzia** [comment]
![Screen Shot 2021-07-14 at 2 57 58 PM](https://user-images.githubusercontent.com/15024880/125698453-cd0d3fa6-f30f-424e-8d5b-c43cda4fa245.png)

Thanks so much for the quick reply. The csv file "CollectedData_Serena.csv' does contain the links to the png files but not the data points. As you can see from the screenshot the data is only for the mac files, not the windows. The mat file also only contains the mac files. Thoughts?

**Comment by MMathisLab** [comment]
are the body parts named differently? If you scroll to the far right in that sheet is the data there; I can confirm everything works for me on merging; namely if you pass `windows2linux=True` it merges data correctly - note, if there is even a space/type wrong, then it appends the bodyparts, as it cannot know how to merge.

**Comment by pkudzia** [comment]
![Screen Shot 2021-07-15 at 10 21 49 AM](https://user-images.githubusercontent.com/15024880/125830463-ba4b3c0b-7034-4872-8979-9af9fbe8c35d.png)
Hi, Thanks for the help. I tried it again from scratch and the CSV file "CollectedData_Serena.csv" still has no data for any of the windows uploaded files. All body parts and naming are the same for all users. The CSV file "CollectedData_SerenaWindwo.csv" does appear to have the labelling data for all of the labelled png files coming from mac and windows labellers. The mat file "CombinedData_Serena95shuffle1.mat' only contains the data from the mac labeller. 

deeplabcut.create_training_dataset(path_config_file, net_type='resnet_50', Shuffles=[SHUF], augmenter_type='imgaug', windows2linux=True)

![Screen Shot 2021-07-15 at 10 25 43 AM](https://user-images.githubusercontent.com/15024880/125830982-804d741e-4ffe-4350-81aa-c96f5407423d.png)

**Comment by AlexEMG** [comment]
I think this happens because you have different labelers. When the pandas arrays are concatenated, they will be concatenated along columns as you have different scorer names (thus, concat by rows doesn't work). When you then create a training set only the scorer name from the config file is used, see here: https://github.com/DeepLabCut/DeepLabCut/blob/master/deeplabcut/generate_training_dataset/trainingsetmanipulation.py#L972
I suppose that name is Serena in your project... 


Anyway, you can fix this by renaming all the annotated data frames to one name. Then you create the training set and you should be fine. You can also do this automatically, we should actually write a cookbook recipe on how to merge datasets from different people!

Here is a helper function that can rename body parts:

```
from deeplabcut.utils import auxiliaryfunctions, auxfun_models, auxfun_multianimal
from pathlib import Path
import os
import pandas as pd

############### FOR RENAMING body parts
def rename_bodyparts(config, pairs):
    """
    Rename bodyparts for list of pairs, e.g. [['snout','nose']] rename snout to nose!
    """

    cfg = auxiliaryfunctions.read_config(config)
    videos = cfg["video_sets"].keys()
    video_names = [Path(i).stem for i in videos]
    folders = [Path(config).parent / "labeled-data" / Path(i) for i in video_names]
    dict_ = {k: v for k, v in pairs}

    for folder in folders:
        filename = os.path.join(str(folder), "CollectedData_" + cfg["scorer"] + ".h5")
        df = pd.read_hdf(filename, "df_with_missing")
        print(df.head())
        df.rename(dict_, level="bodyparts", axis=1, inplace=True)
        df.to_hdf(filename, "df_with_missing",format="table", mode="w")
        df.to_csv(filename.split('.h5')[0]+".csv")

#run function to rename all snouts to noses ... 
rename_bodyparts(path_config_file,  [['snout','nose']])
```

For your case, just adapt 
`df.rename(dict_, level="bodyparts", axis=1, inplace=True)`
to relabel the level with the scorer and make a new function for renaming the scorer. @jeylau let's add something like this to the recipes.

**Comment by pkudzia** [comment]
Ah, thanks so much works like a charm now!  👍  :)

**Comment by MMathisLab** [comment]
@AlexEMG thanks! Also note there is a function that can take edited csv files and convert to h5 plus rename scorer too (so recipe can cover both): deeplabcut.convertcsv2h5('path_to_config.yaml', scorer= 'experimenter')
